test/studies/shootout/meteor/kbrady/meteor-parallel-alt.chpl

```diff
@@ -189,209 +225,242 @@ proc initialise() {
-    rightBorder = leftBorder>>1;
+    rightBorder = leftBorder >> 1;
 
     s1 = (board << 1) | leftBorder;
     s2 = (board >> 1) | rightBorder;
-    s3 = (board << 4) | ((1<<4)-1) | rightBorder;
+    s3 = (board << 4) | ((1 << 4) - 1) | rightBorder;
     s4 = (board >> 4) | leftBorder;
-    s5 = (board << 5) | ((1<<5)-1);
+    s5 = (board << 5) | ((1 << 5) - 1);
     s6 = (board >> 5);
-    s7 = (board << 6) | ((1<<6)-1) | leftBorder;
+    s7 = (board << 6) | ((1 << 6) - 1) | leftBorder;
     s8 = (board >> 6) | rightBorder;
 
-    if ~board&s1&s2&s5&s6 & ((evenRows&s4&s7) | (oddRows&s3&s8)) {
-      return;
-    }
+    if ~board & s1 & s2 & s5 & s6 &
+       ((evenRows & s4 & s7) | (oddRows & s3 & s8)) then return;
 
     count = ctz(~board);
     pos += count;
     board >>= count;
 
-    const f: uint(32) = ((board>>1)&1) | ((board >> (4-(evenRowsLookup[pos] & 1)))&6);
-    const boardAndUsed = board|used;
+    const f = ((board >> 1) & 1) |
+              ((board >> (4 - (evenRowsLookup[pos] & 1))) & 6),
+          boardAndUsed = board | used;
 
-    var mask: uint(32);
-    var currentMask: uint(32) = maskStart[pos][f];
+    var mask: int,
+        currentMask = maskStart[pos][f];
 
     while allMasks[currentMask] {
-      while allMasks[currentMask] & boardAndUsed {
+
+      while allMasks[currentMask] & boardAndUsed do
         currentMask += 1;
-      }
+
       if allMasks[currentMask] {
         mask = allMasks[currentMask];
         currentSolution[placed] = mask;
-        searchLinear(board|(mask & 0x003FFFFF:uint(32)), pos,
-                     used|(mask & 0xFFC00000:uint(32)), placed+1,
-                     currentSolution);
+        searchLinear(board | (mask & maskBottom), pos, used | (mask & maskUsed),
+                     placed + 1, currentSolution);
         currentMask += 1;
       }
     }
   }
 }
 
-proc searchLinearHelper(in board: uint(32), in pos: uint(32),
-                        in used: uint(32), in placed: uint(32),
-                        in firstPiece: uint(32), in mask: uint(32)) {
-  var currentSolution: [0..9] uint(32);
-  currentSolution[0] = firstPiece;
-  currentSolution[1] = mask;
-  begin searchLinear(board, pos, used, placed, currentSolution);
+
+//
+// Lock atomic lock - this is what sync variables do under the hood
+//
+proc lock() {
+  while l.testAndSet() != false do chpl_task_yield();
 }
 
 
-proc searchParallel(in board: uint(32), in pos: uint(32), in used: uint(32),
-                    in placed: uint(32), in firstPiece: uint(32)) {
-  var count: uint(32);
+//
+// Unlock atomic lock
+//
+proc unlock() {
+  l.write(false);
+}
 
-  count = ctz(~board):uint(32);
-  pos += count;
-  board >>= count;
 
-  const boardAndUsed: uint(32) = board|used;
-  var currentMask: uint(32) = maskStart[pos][0];
-  var mask: uint(32);
+//
+// Check a given board as a solution and record it
+//
+proc recordSolution(currentSolution) {
+  var board: [boardDom] int,
+      flipBoard: [boardDom] int,
```

Comment:
There's a few places (such as this one) where you rewrite the type when you can make use of our multivariable declaration style to propagate the type back.  E.g. I'm pretty sure this could be written as `var board, flipBoard: [boardDom] int,`